starters/solidjs-tailwind/src/components/FetchExample/Greeting.spec.jsx

```diff
@@ -1,22 +1,52 @@
-import { render, screen } from 'solid-testing-library';
-import { describe, expect, it, beforeAll, afterAll, afterEach } from 'vitest';
+import { render } from 'solid-testing-library';
+import {
+  describe,
+  expect,
+  it,
+  beforeAll,
+  afterAll,
+  afterEach,
+  beforeEach,
+} from 'vitest';
 import 'whatwg-fetch';
 import { server } from '../../mock/serverSetup';
-import Greeting from './Greeting';
+import { Greeting } from '.';
+import { Router } from '@solidjs/router';
 
 beforeAll(() => server.listen());
 afterAll(() => server.close());
 afterEach(() => server.resetHandlers());
 
 describe('Greeting', () => {
+  let wrapper;
+  beforeEach(() => {
+    wrapper = render(() => (
+      <Router>
+        <Greeting />
+      </Router>
+    ));
+
+    window.fetch = () => {
+      return new Promise((resolve) => {
+        resolve({
+          json: () => {
+            return new Promise((resolve) => {
+              resolve({
+                results: {
+                  text: () => 'daian',
+                },
+              });
+            });
+          },
+        });
+      });
+    };
+  });
   it('should mount', async () => {
-    const wrapper = await render(() => <Greeting />);
     expect(wrapper).toBeTruthy();
   });
 
   it('should show the mocked greeting', async () => {
-    await render(() => <Greeting />);
-    const text = await screen.findByText('Message: Hi Learner');
-    expect(text).toBeVisible();
+    expect(wrapper.findByText('This Dot Labs!')).toBeTruthy();
```

Comment:
This test isn't valid as whatever test you put still passes.

eg
```js
 expect(wrapper.findByText('nothing')).toBeTruthy();
```

Reply:
this should be the test, you notice the await also.

```js
expect(await wrapper.findByText('Hi Learner from This Dot Labs!')).toBeTruthy();
```

As we are mocking the API, you can check the mock folder to see what we mocked